src/uPlot.js

```diff
@@ -1612,51 +1623,80 @@ export default function uPlot(opts, data, then) {
 		}
 
 		// nit: cursor.drag.setSelect is assumed always true
-		if (mouseLeft1 >= 0 && select.show && dragging) {
-			// setSelect should not be triggered on move events
-
-			dragX = drag.x;
-			dragY = drag.y;
+		if (select.show && dragging) {
+			if (src != null) {
+				let [xKey, yKey] = syncOpts.scales;
 
-			let uni = drag.uni;
+				if (xKey) {
+					let sc = scales[xKey];
+					let srcLeft = src.posToVal(src.select[LEFT], xKey);
+					let srcRight = src.posToVal(src.select[LEFT] + src.select[WIDTH], xKey);
 
-			if (uni != null) {
-				let dx = abs(mouseLeft0 - mouseLeft1);
-				let dy = abs(mouseTop0 - mouseTop1);
+					select[LEFT] = getXPos(srcLeft, sc, plotWidCss, 0);
+					select[WIDTH] = abs(select[LEFT] - getXPos(srcRight, sc, plotWidCss, 0));
 
-				dragX = dx >= uni;
-				dragY = dy >= uni;
-
-				// force unidirectionality when both are under uni limit
-				if (!dragX && !dragY) {
-					if (dy > dx)
-						dragY = true;
-					else
-						dragX = true;
+					setStylePx(selectDiv, LEFT, select[LEFT]);
+					setStylePx(selectDiv, WIDTH, select[WIDTH]);
 				}
-			}
 
-			if (dragX) {
-				let minX = min(mouseLeft0, mouseLeft1);
-				let maxX = max(mouseLeft0, mouseLeft1);
-				setStylePx(selectDiv, LEFT,  select[LEFT] = minX);
-				setStylePx(selectDiv, WIDTH, select[WIDTH] = maxX - minX);
+				if (yKey) {
+					let sc = scales[yKey];
+					let srcTop = src.posToVal(src.select[TOP], yKey);
+					let srcBottom = src.posToVal(src.select[TOP] + src.select[HEIGHT], yKey);
 
-				if (uni != null && !dragY) {
-					setStylePx(selectDiv, TOP, select[TOP] = 0);
-					setStylePx(selectDiv, HEIGHT, select[HEIGHT] = plotHgtCss);
-				}
-			}
+					select[TOP] = getYPos(srcTop, sc, plotHgtCss, 0);
+					select[HEIGHT] = abs(select[TOP] - getYPos(srcBottom, sc, plotHgtCss, 0));
 
-			if (dragY) {
-				let minY = min(mouseTop0, mouseTop1);
-				let maxY = max(mouseTop0, mouseTop1);
-				setStylePx(selectDiv, TOP,    select[TOP] = minY);
-				setStylePx(selectDiv, HEIGHT, select[HEIGHT] = maxY - minY);
+					setStylePx(selectDiv, TOP, select[TOP]);
+					setStylePx(selectDiv, HEIGHT, select[HEIGHT]);
+				}
 
-				if (uni != null && !dragX) {
-					setStylePx(selectDiv, LEFT, select[LEFT] = 0);
-					setStylePx(selectDiv, WIDTH, select[WIDTH] = plotWidCss);
+			} else {
+				// setSelect should not be triggered on move events
+	
+				dragX = drag.x;
+				dragY = drag.y;
+	
+				let uni = drag.uni;
+	
+				if (uni != null) {
+					let dx = abs(mouseLeft0 - mouseLeft1);
+					let dy = abs(mouseTop0 - mouseTop1);
+	
+					dragX = dx >= uni;
+					dragY = dy >= uni;
+	
+					// force omnidirectionality when both are under uni limit
```

Comment:
the original comment was correct here, i think. we're forcing *uni*directionality here (not omni) by setting *either* dragX or dragY to true when both are false because they're both under the uni threshold.

Reply:
:+1: